typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts

```diff
@@ -605,11 +619,18 @@ export class SealevelHypSyntheticAdapter extends SealevelHypTokenAdapter {
   }
 
   override async getBalance(owner: Address): Promise<bigint> {
-    const tokenPubKey = this.deriveAssociatedTokenAccount(new PublicKey(owner));
-    const response = await this.getProvider().getTokenAccountBalance(
-      tokenPubKey,
-    );
-    return BigInt(response.value.amount);
+    try {
+      const tokenPubKey = this.deriveAssociatedTokenAccount(
+        new PublicKey(owner),
+      );
```

Comment:
ditto
is there no easy way to dedupe this with the other adapters?

Reply:
There are differences btwn the getBalance implementations, I think we need them all.